SpatialGDK/Source/SpatialGDK/Public/SpatialView/OpList/SplitOpList.h

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "OpList.h"
+#include "Templates/SharedPointer.h"
+
+namespace SpatialGDK
+{
+
+struct SplitOpListData : OpListData
+{
+	// Todo determine if this needs to be thread safe.
```

Comment:
What thread are you worried about here?

Reply:
Threads in the worker connection. Long term it doesn't make a huge amount of sense to have threading in SpatialWorkerConnection while using this, but I hadn't sat down and got my thoughts straight about if different threads would both own this and if so does it matter.

I think by default Unreal makes shared_ptr non thread safe so it was a note to self that I should go back and check this.

+ there is no reason a connection handler might not use this internally, even though currently there is no real reason to use it outside of the worker view.

Reply:
I spent some time working through this. The reference count can only be decremented once it's outside of the method in which it's created, so there isn't really a problem with safety here. It also looks like the reference count is 32 bit so store atomicity isn't an issue and unreal forces all shared pointers to be thread safe on arm.

Reply:
> unreal forces all shared pointers to be thread safe on arm.
Oh I didn't know that, but also not sure why that's relevant - is the reference count a smaller bit size?

Reply:
Arm has a weaker memory model than x86. A 32 bit int store is atomic in both; this was just a quick way to know that "fast" only means something other than what a shared pointer normally means when the cpu already guarantees certain things.

I don't actually think that matters here as the fact that it can only be decremented after creation means relaxed ordering is fine.